openmrs-watcher/src/test/resources/mgt_sender_retry_queue.sql

```diff
@@ -1,14 +1,14 @@
-INSERT INTO sender_retry_queue (id, table_name, primary_key_id, operation, destination, attempt_count, exception_type, message, snapshot, date_created)
+INSERT INTO sender_retry_queue (table_name, primary_key_id, operation, destination, attempt_count, exception_type, message, snapshot, date_created)
 VALUES
-(1, 'person', '1', 'c', 'direct:db-sync', 1, 'java.lang.Exception', 'Testing', 0, '2020-02-27 00:00:00'),
-(2, 'person', '1', 'u', 'direct:db-sync', 1, 'java.lang.Exception', 'Testing', 0, '2020-02-27 00:00:00'),
-(3, 'person', '2', 'c', 'direct:invalid-dest', 1, 'java.lang.Exception', '', 0, '2020-02-27 00:00:00'),
-(4, 'person', '1', 'c', 'direct:senaite', 1, 'java.lang.Exception', null, 0, '2020-02-27 00:00:00'),
-(5, 'orders', '1', 'c', 'mock:event-listener', 1, 'java.lang.Exception', null, 0, '2020-02-27 00:00:00'),
-(6, 'orders', '2', 'c', 'direct:db-sync', 1, 'java.lang.Exception', null, 0, '2020-02-27 00:00:00'),
-(7, 'orders', '101', 'c', 'mock:event-listener', 1, 'java.lang.Exception', 'Testing', 0, '2020-02-27 00:00:00'),
-(8, 'orders', '103', 'c', 'mock:event-listener', 1, 'java.lang.Exception', 'Testing', 0, '2020-02-27 00:00:00'),
-(9, 'test_order', '105', 'c', 'mock:event-listener', 1, 'java.lang.Exception', 'Testing', 0, '2020-02-27 00:00:00'),
-(10, 'drug_order', '107', 'c', 'mock:event-listener', 1, 'java.lang.Exception', 'Testing', 0, '2020-02-27 00:00:00'),
-(11, 'orders', '111', 'c', 'mock:event-listener', 1, 'java.lang.Exception', 'Testing', 0, '2020-02-27 00:00:00'),
-(12, 'referral_order', '113', 'c', 'mock:event-listener', 1, 'java.lang.Exception', 'Testing', 0, '2020-02-27 00:00:00');
+('person', '1', 'c', 'direct:db-sync', 1, 'java.lang.Exception', 'Testing', 0, '2020-02-27 00:00:00'),
+('person', '1', 'u', 'direct:db-sync', 1, 'java.lang.Exception', 'Testing', 0, '2020-02-27 00:00:00'),
+('person', '2', 'c', 'direct:invalid-dest', 1, 'java.lang.Exception', '', 0, '2020-02-27 00:00:00'),
+('person', '1', 'c', 'direct:senaite', 1, 'java.lang.Exception', null, 0, '2020-02-27 00:00:00'),
+('orders', '1', 'c', 'mock:event-listener', 1, 'java.lang.Exception', null, 0, '2020-02-27 00:00:00'),
+('orders', '2', 'c', 'direct:db-sync', 1, 'java.lang.Exception', null, 0, '2020-02-27 00:00:00'),
+('orders', '101', 'c', 'mock:event-listener', 1, 'java.lang.Exception', 'Testing', 0, '2020-02-27 00:00:00'),
+('orders', '103', 'c', 'mock:event-listener', 1, 'java.lang.Exception', 'Testing', 0, '2020-02-27 00:00:00'),
+('test_order', '105', 'c', 'mock:event-listener', 1, 'java.lang.Exception', 'Testing', 0, '2020-02-27 00:00:00'),
+('drug_order', '107', 'c', 'mock:event-listener', 1, 'java.lang.Exception', 'Testing', 0, '2020-02-27 00:00:00'),
+('orders', '111', 'c', 'mock:event-listener', 1, 'java.lang.Exception', 'Testing', 0, '2020-02-27 00:00:00'),
+('referral_order', '113', 'c', 'mock:event-listener', 1, 'java.lang.Exception', 'Testing', 0, '2020-02-27 00:00:00');
```

Comment:
Why do we have to remove the `id` column (the problem being that it makes it possible to introduce duplicate records every time the script is run in tests)?

Reply:
> the problem being that it makes it possible to introduce duplicate records every time the script is run in tests

You can't have duplicate records because `id

Reply:
> the problem being that it makes it possible to introduce duplicate records every time the script is run in tests

The `id` column is the primary key for the table, which means that it is a unique identifier for each record. So, it's impossible to have duplicate records in the table, even if the script is run multiple times. The `id` column is also auto-generated, meaning that it is assigned a unique value automatically when a new record is created. This ensures that there are no conflicts when multiple users are inserting records into the table.

**Note:**
The `DeleteDataTestExecutionListener` listener resets all tables in the database by deleting all rows in them after each test method is executed. This means that all data will be wiped out after each test, which raises the question of whether the above SQL statement is still necessary. Instead, we could potentially utilize the `WatcherTestUtils#addRetryItem()` for retry queue data insertion.

Reply:
Correct that the `id`s autogerated would always be unique. However my point is that we rain explicitly sure that the only records present fo the table are the 12 above. What I was explaining earlier is that removing the `id` column makes it possible to have multiple similar records with just but different `id`s.  Because of this I'd prefer maintaining the `id`s column if there's nothing blocking this.